### DISCUSSION TITLE
How do I get block->id in preview?

### DISCUSSION THREAD

**Discussion by michaeljcoyne** [area17/twill#1987] [General]
I want to be able to access $block->id in the preview. 

I have tried passing it through in block classes but it seems to get reset.

I have a style sheet that relies on having a Id for a block.

I think for some reason Blockable is not accessible here.

See https://github.com/area17/twill/discussions/865

I belive Twill is intentially set up in this way but it is causing me a problem.

Is there a Twill way to do this without trying to hack around or extend the preview class?

**Comment by haringsrob** [comment]
Hey @michaeljcoyne could you share your usecase?

**Reply by michaeljcoyne** [reply]
@haringsrob 
> I have a style sheet that relies on having an Id for a block.

The style sheet gets written with the Id identifier. Its working everywhere but the preview. I can get around this by wrapping the content section in the preview but this wont work for block children.

The block Id seems to be incremental in the preview and not relate to the block ID itself.
